### PR TITLE
CORE-4996 Restructure repository for ledger works

### PR DESCRIPTION
**Pull request by vlajos**
Move old ledger specific code to ledger-obsolete module and prepare ledger:common and ledger:consensual.
The end-goal would be to empty ledger-obsolete on longer term with either deleting what is not needed or salvaging what can be used.
(Most of interesting parts is in the gradle files.)

### CODE REVIEW

File: ledger-obsolete/src/main/kotlin/net/corda/v5/ledger/obsolete/crypto/mocks/DigestServiceMock.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package net.corda.v5.ledger.obsolete.crypto.mocks
```
Comment (blsemo) [inline ledger-obsolete/src/main/kotlin/net/corda/v5/ledger/obsolete/crypto/mocks/DigestServiceMock.kt]: Why is this in `main`?

File: ledger/consensual/build.gradle [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    api 'net.corda.kotlin:kotlin-stdlib-jdk8-osgi'
+    api 'javax.persistence:javax.persistence-api'
+    api 'org.slf4j:slf4j-api'
+
+    compileOnly 'org.osgi:osgi.annotation'
+    // We only need Hibernate for its annotations.
+    compileOnly("org.hibernate:hibernate-osgi:$hibernateVersion") {
+        exclude group: 'org.osgi'
+    }
```
Comment (driessamyn) [inline ledger/consensual/build.gradle]: Can this be removed?
I realise it was in the old API, but hopefully in the new one we can avoid a hibernate dependency and only use `javax.persistence` annotations.

Comment (vlajos) [reply ledger/consensual/build.gradle]: I have removed it. (And the same from common/)

File: ledger-obsolete/src/main/kotlin/net/corda/v5/ledger/obsolete/crypto/mocks/DigestServiceMock.kt (hunk 1 above)
Comment (vlajos) [reply ledger-obsolete/src/main/kotlin/net/corda/v5/ledger/obsolete/crypto/mocks/DigestServiceMock.kt]: I have moved it to test/

**Review by blsemo — DISMISSED** [review]

**Review by benmcmahon100 — DISMISSED** [review]

File: gradle.properties [hunk 3]
```diff
@@ -7,7 +7,7 @@ cordaProductVersion = 5.0.0
 # NOTE: update this each time this module contains a breaking change
 ## NOTE: currently this is a top level revision, so all API versions will line up, but this could be moved to
 ##   a per module property in which case module versions can change independently.
-cordaApiRevision = 120
+cordaApiRevision = 121
```
Comment (driessamyn) [inline gradle.properties]: Beware there is already a 121 approved: https://github.com/corda/corda-api/pull/425
and will be merged when @JamesHR3's duplicate 120 is backed out, so probably best to synch up with @JamesHR3 in terms of who merges first.

**Review by driessamyn — DISMISSED** [review]
Please hold off merging till the version issues are resolved.

File: gradle.properties (hunk 3 above)
Comment (vlajos) [reply gradle.properties]: @driessamyn @JamesHR3 
I am happy to wait with this.
(Please ping me when that one has been merged.)

Comment (driessamyn) [reply gradle.properties]: 121 is merged and build is back green: https://ci02.dev.r3.com/blue/organizations/jenkins/Corda5%2Fcorda-api/detail/release%2Fos%2F5.0/766/pipeline/104

Comment (vlajos) [reply gradle.properties]: @driessamyn 
Thanks. I have just rebased on the top of latest release/os/5.0 and incremented the version.

Comment (driessamyn) [reply gradle.properties]: Thanks!
Good for me.

File: ledger-obsolete/src/test/java/net/corda/v5/ledger/obsolete/contracts/TransactionVerificationExceptionJavaApiTest.java [hunk 4]
```diff
@@ -1,31 +1,32 @@
...
+import net.corda.v5.ledger.obsolete.contracts.TransactionVerificationException.TransactionRequiredContractUnspecifiedException;
+import net.corda.v5.ledger.obsolete.contracts.TransactionVerificationException.TransactionGroupParameterOrderingException;
+import net.corda.v5.ledger.obsolete.contracts.TransactionVerificationException.MissingGroupParametersException;
+import net.corda.v5.ledger.obsolete.contracts.TransactionVerificationException.BrokenTransactionException;
+import net.corda.v5.ledger.obsolete.contracts.TransactionVerificationException.InvalidAttachmentException;
+import net.corda.v5.ledger.obsolete.contracts.TransactionVerificationException.UnsupportedClassVersionError;
+import net.corda.v5.ledger.obsolete.contracts.TransactionVerificationException.UntrustedAttachmentsException;
+import net.corda.v5.ledger.obsolete.contracts.TransactionVerificationException.UnsupportedHashTypeException;
+import net.corda.v5.ledger.obsolete.identity.Party;
+import net.corda.v5.ledger.obsolete.contracts.*;
```
Comment (blsemo) [inline ledger-obsolete/src/test/java/net/corda/v5/ledger/obsolete/contracts/TransactionVerificationExceptionJavaApiTest.java]: Wild card imports are not allowed? How did this pass detekt?

File: ledger-obsolete/src/test/java/net/corda/v5/ledger/obsolete/transactions/LedgerTransactionJavaApiTest.java [hunk 5]
```diff
@@ -1,19 +1,22 @@
...
+import net.corda.v5.ledger.obsolete.contracts.StateAndRef;
+import net.corda.v5.ledger.obsolete.contracts.StateRef;
+import net.corda.v5.ledger.obsolete.contracts.TimeWindow;
+import net.corda.v5.ledger.obsolete.contracts.TransactionState;
+import net.corda.v5.ledger.obsolete.identity.AbstractParty;
+import net.corda.v5.ledger.obsolete.identity.Party;
+import net.corda.v5.ledger.obsolete.identity.AbstractParty;
+import net.corda.v5.ledger.obsolete.identity.Party;
+import net.corda.v5.ledger.obsolete.transactions.LedgerTransaction;
 import org.assertj.core.api.Assertions;
```
Comment (blsemo) [inline ledger-obsolete/src/test/java/net/corda/v5/ledger/obsolete/transactions/LedgerTransactionJavaApiTest.java]: There are duplicate imports - can you optimise imports, please?

File: ledger-obsolete/src/test/java/net/corda/v5/ledger/obsolete/transactions/TransactionBuilderJavaApiTest.java [hunk 6]
```diff
@@ -1,19 +1,22 @@
...
+import net.corda.v5.ledger.obsolete.contracts.ReferencedStateAndRef;
+import net.corda.v5.ledger.obsolete.contracts.StateAndRef;
+import net.corda.v5.ledger.obsolete.contracts.StateRef;
+import net.corda.v5.ledger.obsolete.contracts.TimeWindow;
+import net.corda.v5.ledger.obsolete.contracts.TransactionState;
+import net.corda.v5.ledger.obsolete.identity.AbstractParty;
+import net.corda.v5.ledger.obsolete.identity.Party;
+import net.corda.v5.ledger.obsolete.identity.AbstractParty;
+import net.corda.v5.ledger.obsolete.identity.Party;
+import net.corda.v5.ledger.obsolete.transactions.*;
```
Comment (blsemo) [inline ledger-obsolete/src/test/java/net/corda/v5/ledger/obsolete/transactions/TransactionBuilderJavaApiTest.java]: No wild card imports

File: ledger-obsolete/src/test/java/net/corda/v5/ledger/obsolete/contracts/StatePointersJavaApiTest.java [hunk 7]
```diff
@@ -1,7 +1,9 @@
-package net.corda.v5.ledger.contracts;
+package net.corda.v5.ledger.obsolete.contracts;
 
 import net.corda.v5.crypto.SecureHash;
-import net.corda.v5.ledger.UniqueIdentifier;
+import net.corda.v5.ledger.obsolete.UniqueIdentifier;
+import net.corda.v5.ledger.obsolete.UniqueIdentifier;
+import net.corda.v5.ledger.obsolete.contracts.*;
```
Comment (blsemo) [inline ledger-obsolete/src/test/java/net/corda/v5/ledger/obsolete/contracts/StatePointersJavaApiTest.java]: No wild card imports

**Review by blsemo — COMMENTED** [review]
The Java API tests for ledger.obsolete have messed up imports (see comments for examples), with duplicate imports and wild card imports.

**Review by blsemo — APPROVED** [review]
LGTM

**Review by benmcmahon100 — APPROVED** [review]

**Pull request merged**